### PR TITLE
New Shares Spice

### PR DESCRIPTION
**Pull request by skrobek**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer: **`New Shares Spice`**

###### Description of changes

Currently, the Stock Instant Answer only supports US ticker symbols and returns a small quantity of data. With a Stocks (International) we are able to support searches by RIC code and display valuable information i.e. PE Ratio, Dividend yield.

---

Instant Answer Page: https://duck.co/ia/view/shares

@skrobek

### CODE REVIEW

**Comment by daxtheduck** [comment]
@skrobek Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by edgesince84** [comment]
thanks @skrobek - will check this out soon!

File: lib/DDG/Spice/Shares.pm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package DDG::Spice::Shares;
+# ABSTRACT: Stock quote ticker
+
+use strict;
+use DDG::Spice;
+
+triggers startend => 'stock quote for', 'stock quote', 'stock price of', 'stock price', 'share price', 'stocks', 'shares', 'share price for', 'share price of';
+
+spice to => 'https://og6c1r82zb.execute-api.us-east-1.amazonaws.com/v1/ddg?term=$1';
```
Comment (moollaza) [inline lib/DDG/Spice/Shares.pm]: Is this a custom Stockflare API endpoint for DDG?

**Comment by moollaza** [comment]
@skrobek thanks for making this! is this intention to have this replace the existing Stocks IA?

File: lib/DDG/Spice/Shares.pm (hunk 1 above)
Comment (skrobek) [inline lib/DDG/Spice/Shares.pm]: Yes, that's custom Stockflare API endpoint.

Comment (stratmm) [inline lib/DDG/Spice/Shares.pm]: Hi, @moollaza 

Yes we created this endpoint just for DDG, it is simply a AWS API-Gateway endpoint triggering a Lambda Function that in turn uses our internal APIs.  We decided to keep it separate from the api that serves our site `https://api.stockflare.com` just for simplicity sake.

We can easily put it on a custom domain using our certificate if you prefer, something like ddg.stockflare.com

**Comment by stratmm** [comment]
Hi @moollaza,

We deliberately called this `Shares` so it could co-exist with `Stocks`.  We do have the advantage that we have coverage of all stocks globally and deliver more data, but we would have no problems with both `Stocks` and `Shares`  both being shown for US stocks.

File: lib/DDG/Spice/Shares.pm (hunk 1 above)
Comment (moollaza) [inline lib/DDG/Spice/Shares.pm]: @stratmm thanks for clarifying. This should be fine, I just wanted to confirm, this wasn't something unofficial or illegitimate 👍

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=share price fb.o)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/shares_mobile.png?FEKFZAmfX2)

**Comment by moollaza** [comment]
@skrobek thanks for clarifying. 

Our policy is that Instant Answers providing the same/similar info shouldn't have overlapping query spaces. Right now it looks like these will overlap for certain queries, but that could be addressed (e.g. only use Stockflare for RIC code searches or P/E Ratio, Market Value queries.)

I'll also need to check with our BD team if this IA can move forward because I believe NASDAQ data may require some special handling.

I'll get back to you shortly with more info.

Thanks again!

**Comment by stratmm** [comment]
@moollaza Thanks for the update.  Just thought that I would clarify how our search term works in case that helps.

Our api takes the search term and does up to three queries in the following order to find a hit:
1.  Searches for a matching RIC;
2.  Searches for a matching ticker;
3.  Finally performs a textual search across company name and description.

So for instance `share price walgreens` will yield a direct hit on `Walgreens Boots Alliance Inc` as well as giving a hit on the RIC WBA.O and the ticker WBA. 

We also do a bit of other optimisation based on the search term length and number of words so for instance a search like `share price colombia banking` will still give you a good hit like `Bancolombia SA (ADR)`

Thanks
Mark

**Comment by shaneleonard121** [comment]
@moollaza you are correct that there are special licensing rules for all stock exchange data. For example displaying real-time data from Nasdaq or NYSE requires a per user license OR a $500k p.a. media license. You will note that we are not providing real-time. Some websites get around this issue by displaying the latest-price from BATS Markets instead of the price from Nasdaq or NYSE, which is probably what Xignite is doing for you on US equities.

We in contrast are providing end-of-day data, for which our data vendor, Thomson Reuters has negotiated a blanked right to distribute license free will all stock exchanges globally. That is why we are able to deliver a complete globally solution to our users and provide you with an Instant Answer solution on a global basis. 

I hope that helps clarify your Nasdaq question. All the best.

**Comment by shaneleonard121** [comment]
@moollaza is there any additional information you'd like from us, to help move this forward? Thanks in advance. Shane

**Comment by moollaza** [comment]
@shaneleonard121 apologies for the delay and thanks for all the explaining things in more detail.

Right now DuckDuckHack is in the midst of a big change: we're narrowing our focus for Instant Answers with a new goal of becoming the best search engine for Programmers.

This means we're now dedicating our focus entirely towards Instant Answers related to our Programming Mission. 

Unfortunately that means this IA is outside the scope of our focus so we can't dedicate any time to reviewing it. Our BD team would also need to review our current agreements with Xignite to make sure this Instant Answer is acceptable.

If our focus changes in the future, I'd be happy to reconsider how DuckDuckGo and StockFlare can work together.

Thanks a lot for taking the time to contribute, sorry we can't currently accept your pull request.

**Pull request closed without merging**